### ISSUE TITLE
[Form] Inline message with validation message is not announced right by screen readers

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3733]
### 😯 Current Behavior

When inline message gets displayed, screen reader announces all the fields with errors without any spaces between their names. 
This is hard to understand when there are multiple names.

(Turn volume on to experience this bug 🔈)

https://github.com/user-attachments/assets/f3481156-9f64-4fc9-aab6-b81fc78c48f5

### 🤔 Expected Behavior/possible solution?

This error could have two potential root causes. 

1. String that is passed to `sl-announcer` does not include spaces between each field name. 
2. Because each field name is rendered as a link, sl-announcer may announce the message incorrectly. 

Only plain text with fields' names should be passed to `sl-announcer`. After each field's name there should be space, so each name is a standalone name.

### 🖥️ Steps to Reproduce

1. Go to https://storybook.sanomalearning.design/?path=/story/examples-form--composite-form
2. Leave all fields empty
3. With screen reader turned on submit the form
4. Check if error message announced by screen readers is easy to understand

### Were you able to reproduce this in our storybook?

Yes

### If yes, provide the storybook URL

https://storybook.sanomalearning.design/?path=/story/examples-form--composite-form

### ⚖️ Impact

This blocks our release or stops us from using this component

### Framework

Lit

### What browsers are you seeing the problem on?

All

### Component version

1.6.0

### 👤 Your name

Kacper

### 🧢 Your product/team

SLDS

### 🎨 Your Theme(s)

Sanoma Learning
